chips/stm32f303xc/src/adc.rs

```diff
@@ -623,6 +633,31 @@ impl Adc {
     pub fn disable_clock(&self) {
         self.clock.disable();
     }
+
+    fn enable_special_channels(&self) {
+        // enabling temperature channel
+        if self.requeste_channel.get() == 16 {
+            self.enable_temperature();
+        }
+    }
+
+    fn sample_u32(&self, channel: u32) -> ReturnCode {
+        self.enable_special_channels();
+        if self.status.get() == ADCStatus::Idle {
+            self.requested.set(ADCStatus::Idle);
+            self.status.set(ADCStatus::OneSample);
+            self.registers.smpr2.modify(SMPR2::SMP16.val(0b100));
+            self.registers.sqr1.modify(SQR1::L.val(0b0000));
+            self.registers.sqr1.modify(SQR1::SQ1.val(channel));
+            self.registers.ier.modify(IER::EOSIE::SET);
+            self.registers.ier.modify(IER::EOCIE::SET);
+            self.registers.ier.modify(IER::EOSMPIE::SET);
+            self.registers.cr.modify(CR::ADSTART::SET);
+            ReturnCode::SUCCESS
+        } else {
+            ReturnCode::EBUSY
+        }
+    }
```

Comment:
Its not intuitive to me why a function named `sample_u32` enables the temperature channel any time it is called